### PR TITLE
Dcka 838 did managament

### PR DESCRIPTION
**Pull request by osinakayah**

### CODE REVIEW

File: src/features/didManagement/CreateNewDIDScreen.js [hunk 1]
```diff
@@ -54,76 +54,85 @@ export function CreateNewDIDScreen() {
...
-        </FormControl>
-        <FormControl>
+      <ScrollView marginLeft={3} marginRight={3}>
+        <FormControl isInvalid={form._errors.didType}>
           <Stack mt={7}>
             <FormControl.Label>
               {translate('didManagement.did_type')}
             </FormControl.Label>
-            <Select>
+            <Select
```
Comment (mike-parkhill) [inline src/features/didManagement/CreateNewDIDScreen.js:63]: does this need a test ID?

File: src/features/didManagement/CreateNewDIDScreen.test.js [hunk 2]
```diff
@@ -1,17 +1,75 @@
...
+        handleSubmit={jest.fn()}
+      />,
+      {
+        context: {store: mockStore(initialState)},
+      },
+    );
+    expect(wrapper.dive()).toMatchSnapshot();
+  });
+
+  it('should render correctly with hmm', () => {
```
Comment (mike-parkhill) [inline src/features/didManagement/CreateNewDIDScreen.test.js:67]: `with hmm`? 😕

File: src/features/didManagement/DIDListScreen.js [hunk 3]
```diff
@@ -48,45 +54,73 @@ export function DIDListScreen() {
...
-              size={'xs'}>
-              <Typography
-                variant="transaction-filter"
-                color={Theme.colors.inactiveText}>
-                {translate('didManagement.learn_more')}
+      <ScrollView marginLeft={3} marginRight={3}>
+        {didList.length > 0 ? (
+          didList.map(didDocumentResolution => {
+            return (
+              <DIDListItem
```
Comment (mike-parkhill) [inline src/features/didManagement/DIDListScreen.js:61]: do these need testIds so they can be easily selected?

File: src/features/didManagement/components/SingleDIDOptionsModal.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+  const content = (
+    <Stack p={8} testID="addAccountModal">
+      <Typography variant="h1">
+        {translate('didManagement.edit_did')}
+      </Typography>
+      <OptionList
+        mt={5}
+        items={[
+          // {
```
Comment (mike-parkhill) [inline src/features/didManagement/components/SingleDIDOptionsModal.js:38]: is this placeholder for when you implement edit?

File: src/features/didManagement/didManagment-slice.js [hunk 5]
```diff
@@ -47,24 +47,43 @@ const initializeDID = async () => {
...
+      await createDefaultDID();
     } catch (e) {
       captureException(e);
     }
   }
 };
 
+const createDockDID = () => {};
+export const createNewDID = async newDidParams => {
+  if (newDidParams.didType === 'didkey') {
```
Comment (mike-parkhill) [inline src/features/didManagement/didManagment-slice.js:71]: I generally dislike `if` trees like this. They're not bad when there's only two options, but get unwieldy very quickly as you add more. A pattern I like is to register factory functions in a dictionary using the type as a key.  This function would then just be a dictionary lookup.

    const didFactories = { 
         'diddock': createDockDID, 
         'didkey', createKeyDID
    };

Probably means refactoring to pass newDidParams into all the factory functions but that's probably okay.

File: src/features/didManagement/didManagment-slice.js [hunk 6]
```diff
@@ -47,24 +47,43 @@ const initializeDID = async () => {
...
-      await initializeDID();
+      await createDefaultDID();
     } catch (e) {
       captureException(e);
     }
   }
 };
 
+const createDockDID = () => {};
+export const createNewDID = async newDidParams => {
```
Comment (mike-parkhill) [inline src/features/didManagement/didManagment-slice.js:70]: are there tests that can be written to make sure you get the right DID being generated? 

Not sure how we'd manage the did:dock blockchain access.

File: src/locales/languages/en.json [hunk 7]
```diff
@@ -324,6 +324,16 @@
     "share_did_desc": "Share your Decentralized Identifier",
     "did_dock_info": "did:dock requires DOCK token to create. This is used for the transaction fee required to create the DID and have a record on the blockchain.",
     "enter_password": "Enter Password",
-    "confirm_password": "Confirm Password"
+    "confirm_password": "Confirm Password",
+    "share": "Share",
+    "did_created": "DID Created",
+    "did_deleted": "DID Deleted Successfully",
+    "did_creation_error": "Oops, an error occurred creating DID",
```
Comment (mike-parkhill) [inline src/locales/languages/en.json:331]: not sure about the `Oops`. Don't think we use that casual tone elsewhere.

File: src/features/didManagement/didManagment-slice.js (hunk 6 above)
Comment (osinakayah) [reply src/features/didManagement/didManagment-slice.js:70]: > Not sure how we'd manage the did:dock blockchain access.
Yes, I will include the test for did:key for now.

File: src/core/redux-store.js [hunk 8]
```diff
@@ -11,7 +11,7 @@ let store;
 const middleware = [thunk];
 
 if (NODE_ENV === 'development') {
-  middleware.push(logger);
+  // middleware.push(logger);
```
Comment (mike-parkhill) [inline src/core/redux-store.js:14]: did you intend to commit this?

File: src/features/didManagement/didHooks.js [hunk 9]
```diff
@@ -54,14 +59,27 @@ export function useDIDManagementHandlers() {
         type: 'error',
       });
     }
-  }, [form, queryDIDDocuments]);
+  }, [form]);
+
+  const onEditDID = useCallback(async () => {
+    const {id, didName} = form;
+    await updateDIDDocument(id, {
+      name: didName,
```
Comment (mike-parkhill) [inline src/features/didManagement/didHooks.js:67]: how about trim here too? Or is there somewhere on the form that we can trim once so we don't have to always check it?

File: src/features/didManagement/didHooks.js [hunk 10]
```diff
@@ -70,6 +88,7 @@ export function useDIDManagement() {
     const didResolutionDocuments = await wallet.query({
       type: 'DIDResolutionResponse',
     });
+    console.log(didResolutionDocuments.length, 'didResolutionDocuments.length');
```
Comment (mike-parkhill) [inline src/features/didManagement/didHooks.js:91]: did you intend to leave this?

**Review by mike-parkhill — CHANGES_REQUESTED** [review]
Some linting errors too.

File: src/features/didManagement/didHooks.js [hunk 11]
```diff
@@ -36,83 +30,75 @@ export function useDIDManagementHandlers() {
...
+
+  const onDeleteDID = useCallback(
+    async didResolution => {
+      try {
+        await deleteDID({id: didResolution.id});
+        showToast({
+          message: translate('didManagement.did_deleted'),
+          type: 'success',
+        });
+      } catch (e) {}
```
Comment (mike-parkhill) [inline src/features/didManagement/didHooks.js:66]: should we be doing something with this error?

File: src/features/didManagement/didHooks.js [hunk 12]
```diff
@@ -36,83 +30,75 @@ export function useDIDManagementHandlers() {
...
       });
       navigate(Routes.DID_MANAGEMENT_LIST);
       logAnalyticsEvent(ANALYTICS_EVENT.DID.DID_CREATED, {});
     } catch (e) {
-      showToast({
-        message: translate('didManagement.did_creation_error'),
-        type: 'error',
-      });
+      throw new Error(
+        translate('didManagement.error_keypair_type', {
```
[the remark this answers is not in the document]
Comment (osinakayah) [reply src/features/didManagement/didHooks.js:51]: Hmm, true. 
We have a helper function that reads the actual error message, displays them and post them to sentry, let me it instead.

File: src/locales/languages/en.json [hunk 13]
```diff
@@ -329,13 +329,14 @@
     "did_created": "DID Created",
     "did_deleted": "DID Deleted Successfully",
     "did_creation_error": "An error occurred creating DID",
-    "error_keypair_type": "%{keypairType} is not supported.",
+    "did_updating_error": "An error occurred updating DID",
+    "error_keypair_type": "%{keypairType} keypair type  is not supported.",
```
Comment (mike-parkhill) [inline src/locales/languages/en.json:333]: here's the other half of that double space 😄

**Review by mike-parkhill — APPROVED** [review]

**Pull request merged**